tests/cnf/core/network/policy/tests/multinetpolicy_bondoversriov.go

```diff
@@ -0,0 +1,729 @@
+
+	_, err = nad.NewBuilder(APIClient, nadName, nsName).WithMasterPlugin(config).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create net-attach-def")
+}
+
+func defineAndCreatePodWithBondIf(podName, nsName, net1Name, net2Name string, testData podsData) *pod.Builder {
+	var rootUser int64
+
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW", "NET_ADMIN"},
+		},
+	}
+
+	netAnnotation := []*types.NetworkSelectionElement{
+		{
+			Name:             net1Name,
+			InterfaceRequest: "net1",
+		},
+		{
+			Name:             net2Name,
+			InterfaceRequest: "net2",
+		},
+		{
+			Name:             "bond",
+			InterfaceRequest: "bond1",
+			IPRequest:        append(append([]string{}, testData[podName].IPv4), testData[podName].IPv6),
+		},
+	}
+
+	tPodBuilder := pod.NewBuilder(APIClient, podName, nsName, NetConfig.CnfNetTestContainer).
+		WithNodeSelector(map[string]string{"kubernetes.io/hostname": "worker-0"}).
+		WithSecondaryNetwork(netAnnotation).
+		WithPrivilegedFlag().
+		WithLabel("app", podName)
+
+	for index, protocolPort := range testData[podName].ProtocolPort {
+		containerBuilder, err := pod.NewContainerBuilder(strings.ReplaceAll(protocolPort, ":", ""),
+			NetConfig.CnfNetTestContainer,
+			[]string{"/bin/bash", "-c", fmt.Sprintf("testcmd -listen -interface bond1 -protocol %s -port %s",
+				strings.Split(protocolPort, ":")[0], strings.Split(protocolPort, ":")[1])}).
+			WithSecurityContext(&securityContext).
+			GetContainerCfg()
+		Expect(err).ToNot(HaveOccurred(), "Failed to get container config")
+
+		if index == 0 {
+			tPodBuilder.RedefineDefaultContainer(*containerBuilder)
+		} else {
+			tPodBuilder.WithAdditionalContainer(containerBuilder)
+		}
+	}
+
+	tPod, err := tPodBuilder.CreateAndWaitUntilRunning(1 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create test pod")
+
+	return tPod
+}
+
+func defineNmapValidPaths(sPods, dPods []*pod.Builder, expectedResult []int64, testData podsData) {
+	By("Deriving applicable paths between given source and destination pods")
+
+	resultIndex := 0
+
+	for _, sPod := range sPods {
+		for _, dPod := range dPods {
+			var ipv4host, ipv6host string
+
+			// Convert expectedResult to Binary format string and do zero padding if the length of binary string is not
+			// equal to number of available paths. i.e., twice (ipv4 and ipv6) the no. of ports a pod listens to.
+			resultBinary := strconv.FormatInt(expectedResult[resultIndex], 2)
+			if len(resultBinary) < 2*len(testData[dPod.Object.Name].ProtocolPort) {
+				resultBinary = strings.Join([]string{strings.Repeat("0",
+					2*len(testData[dPod.Object.Name].ProtocolPort)-len(resultBinary)), resultBinary}, "")
+			}
+
+			resultIndex++
+
+			nmapPorts, nmapProtocols := defineNmapPortsProtocols(testData[dPod.Object.Name].ProtocolPort)
+
+			if ipv4host != testData[dPod.Object.Name].IPv4 {
```

Comment:
This condition is always true, as `ipv4Host` is empty at every loop.

I'm not getting the meaning here, as it looks quite complicated to me.